### PR TITLE
Allow to have no macros in Compiler

### PR DESCRIPTION
**Pull request by f3l1x**
- bug fix: #176 
- BC break? no
- doc PR: no needed

This PR has 2 changes:
- upgrade nette/tester to v2.0
- allow to use own Compiler with no defined macros

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
 		"ext-tokenizer": "*"
 	},
 	"require-dev": {
-		"nette/tester": "~1.7",
+		"nette/tester": "^2.0.1",
```
[the remark this answers is not in the document]
Comment (f3l1x) [reply composer.json]: Yop, my mistake.

File: src/Latte/Compiler/Compiler.php [hunk 2]
```diff
@@ -124,7 +124,7 @@ public function compile(array $tokens, $className)
 		$this->methods = ['main' => null, 'prepare' => null];
 
 		$macroHandlers = new \SplObjectStorage;
-		array_map([$macroHandlers, 'attach'], call_user_func_array('array_merge', $this->macros));
+		array_map([$macroHandlers, 'attach'], $this->macros ? call_user_func_array('array_merge', $this->macros): []);
```
[the remark this answers is not in the document]
Comment (f3l1x) [reply src/Latte/Compiler/Compiler.php]:
```
Exited with error code 255 (expected 0)
   E_WARNING: array_merge() expects at least 1 parameter, 0 given
```

File: src/Latte/Compiler/Compiler.php [hunk 3]
```diff
@@ -124,7 +124,10 @@ public function compile(array $tokens, $className)
 		$this->methods = ['main' => null, 'prepare' => null];
 
 		$macroHandlers = new \SplObjectStorage;
-		array_map([$macroHandlers, 'attach'], call_user_func_array('array_merge', $this->macros));
+
+		if ($this->macros) {
```
Comment (dg) [inline src/Latte/Compiler/Compiler.php]: Is condition needed?

Comment (f3l1x) [reply src/Latte/Compiler/Compiler.php]: Yep, it triggers

```
Exited with error code 255 (expected 0)
   E_WARNING: array_merge() expects at least 1 parameter, 0 given
```

Comment (dg) [reply src/Latte/Compiler/Compiler.php]: Ok, good.

**Comment by dg** [comment]
Please fix coding style errors and I'll merge it.

**Comment by f3l1x** [comment]
Thanks.

**Pull request merged**